spec/views/users/backup_code_setup/create.html.erb_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+require 'data_uri'
+
+describe 'users/backup_code_setup/create.html.erb' do
+  let(:user) { build(:user, :signed_up) }
+
+  module URI
+    def self.decode(value)
+      CGI.unescape(value)
+    end
+  end
+
+  before do
+    allow(view).to receive(:current_user).and_return(user)
+    @codes = BackupCodeGenerator.new(user).create
+  end
+
+  it 'displays download link with plain text content equal to the users backup codes' do
+    render
+
+    doc = Nokogiri::HTML(rendered)
+    download_link = doc.at_css('a[download]')
+    data_uri = URI::Data.new(download_link[:href])
```

Comment:
Huh, I didn't know we had this gem. Quite handy! Could replace the need for our [`DataUrlImage` class](https://github.com/18F/identity-idp/blob/main/app/services/idv/data_url_image.rb) too, though after [recent optimizations](https://github.com/18F/identity-idp/pull/6413) and some local benchmarking, I think our implementation still performs a bit better.

Reply:
I think @zachmargolis deserves the credit for this one. He added it when he moved personal key downloads client side.